src/PoolManager.sol

```diff
@@ -185,16 +185,20 @@ contract PoolManager is IPoolManager, Fees, NoDelegateCall, Claims {
         IPoolManager.ModifyPositionParams memory params,
         bytes calldata hookData
     ) external override noDelegateCall onlyByLocker returns (BalanceDelta delta) {
+        PoolId id = key.toId();
+        if (pools[id].isNotInitialized()) revert PoolNotInitialized();
+
         if (key.hooks.shouldCallBeforeModifyPosition()) {
-            if (
-                key.hooks.beforeModifyPosition(msg.sender, key, params, hookData)
-                    != IHooks.beforeModifyPosition.selector
-            ) {
-                revert Hooks.InvalidHookResponse();
+            bytes4 selector = key.hooks.beforeModifyPosition(msg.sender, key, params, hookData);
+            if (selector != IHooks.beforeModifyPosition.selector) {
+                if (key.hooks.isValidNoOpCall(selector)) {
```

Comment:
can flatten these ifs with
```
if (IisValidNoop(selector) => return 0
else if (selector != correct selector) => revert

```

Reply:
downside is one extra branch and check in the happy not-noop path but personally think thats worth the flattening